src/d/actor/d_a_btd.cpp

```diff
@@ -1,208 +1,2550 @@
+            }
+            local_28 = apdVar1->eyePos - i_this->eyePos;
+            uVar2 = cM_atan2s(local_28.x, local_28.z) - i_this->current.angle.y;
+            target = ((short)uVar2 / 2);
+            if (target > 0x2000) {
+                target = 0x2000;
+            } else if (target < -0x2000) {
+                target = -0x2000;
+            }
+        }
+    } else {
+        target = 0;
+        sVar4 = 0;
+    }
+    cLib_addCalcAngleS2(&i_this->m5E88, sVar4, 4, 0x200);
+    cLib_addCalcAngleS2(&i_this->m5E8E, target, 4, 0x200);
+    cLib_addCalcAngleS2(&i_this->m5E8A, i_this->m5E8C, 4, 0x100);
+    i_this->m5E86 = 0;
+    i_this->m5E8C = 0;
 }
 
 /* 00005BCC-00005C24       .text k_a_d_sub__FPvPv */
-void k_a_d_sub(void*, void*) {
-    /* Nonmatching */
+void* k_a_d_sub(void* i_this, void* param_2) {
+    if (((fopAc_IsActor(i_this)) && (fopAcM_GetName(i_this) == PROC_KUI)) && (fopAcM_GetParam(i_this) == 0x511)) {
+        return i_this;
+    }
+    return param_2 = NULL;
 }
 
 /* 00005C24-00005C70       .text dr2_a_d_sub__FPvPv */
-void dr2_a_d_sub(void*, void*) {
-    /* Nonmatching */
+void* dr2_a_d_sub(void* i_this, void* param_2) {
+    if ((fopAc_IsActor(i_this)) && (fopAcM_GetName(i_this) == PROC_DR2)) {
+        return i_this;
+    } else {
+        return param_2 = NULL;
+    }
 }
 
 /* 00005C70-00005CC8       .text wepon_s_sub__FPvPv */
-void wepon_s_sub(void*, void*) {
-    /* Nonmatching */
+void* wepon_s_sub(void* i_this, void* param_2) {
+    if (((fopAc_IsActor(i_this)) && (fopAcM_GetName(i_this) == PROC_HIMO2)) || (fopAcM_GetName(i_this) == PROC_BOOMERANG)) {
+        return i_this;
+    }
+    return param_2 = NULL;
 }
 
 /* 00005CC8-00005DC4       .text hahen_move__FP9btd_class */
-void hahen_move(btd_class*) {
-    /* Nonmatching */
+void hahen_move(btd_class* i_this) {
+    btd_hahen_s* pbVar2 = &i_this->hahen[0];
+    for (s32 i = 0; i < (s32)ARRAY_SIZE(i_this->hahen); i++, pbVar2++) {
+        if (pbVar2->m04 != 0) {
+            pbVar2->m08 += pbVar2->m14;
+            pbVar2->m20 += 0x900;
+            pbVar2->m22 += 0x600;
+            pbVar2->m14.y -= 3.0f;
+            MtxTrans(pbVar2->m08.x, pbVar2->m08.y, pbVar2->m08.z, false);
+            mDoMtx_YrotM(*calc_mtx, pbVar2->m22);
+            mDoMtx_XrotM(*calc_mtx, pbVar2->m20);
+            pbVar2->model->setBaseTRMtx(*calc_mtx);
+            if (pbVar2->m08.y < -100.0f) {
+                pbVar2->m04 = 0;
+            }
+        }
+    }
 }
 
 /* 00005DC4-000060E0       .text sibuki_move__FP9btd_class */
-void sibuki_move(btd_class*) {
-    /* Nonmatching */
+void sibuki_move(btd_class* i_this) {
+    /* Nonmatching - regalloc */
+    daPy_py_c* apdVar4 = daPy_getPlayerActorClass();
+    btd_sibuki_s* sibuikiS = &i_this->mBtdSibukiS[0];
+    for (s32 i = 0; i < (s32)ARRAY_SIZE(i_this->mBtdSibukiS); i++, sibuikiS++) {
+        if (sibuikiS->m00 != 0) {
+            if ((char)sibuikiS->m00 <= 2) {
+                bool bVar2 = false;
```

Comment:
Does this still work with a continue?
```suggestion
        if (sibuikiS->m00 == 0) {
            continue;
        }
        if ((char)sibuikiS->m00 <= 2) {
            bool bVar2 = false;
```